changes/2020-07-14_refactor-cmc-spec/change.md

```diff
@@ -0,0 +1,247 @@
+on every Get Cache Entry and Put Cache Entry operation,
+and also that the it evicts any additional entries as necessary
+to not exceed the entry capacity
+on every Put Cache Entry operation.
+This "lazy eviction" is sufficient to ensure that the local CMC
+neither returns TTL-expired entries
+nor exceeds the entry capacity.
+
+The change also recommends that the local CMC evict TTL-expired entries on a periodic basis,
+if the implementation's programming language and runtime make doing so feasible.
+By doing so,
+the local CMC ensures that no entry lives for more than a constant amount of time past its TTL;
+without periodic eviction,
+a TTL-expired entry could remain in the cache indefinitely
+if no user makes any Get Cache Entry or Put Cache Entry call
+after the entry enters the cache.
+The change does not mandate period eviction because
+concurrency is complex to implement in some programming languages and runtimes.
+
+This change specifies that the local CMC
+evicts entries in least recently used (LRU) order.
+Prior to this change,
+the specification already stated
+that the local CMC is a least recently used LRU cache.
+Since LRU eviction is a sane default strategy for many use cases,
+and since LRU cache implementations exist in most mainstream programming language ecosystems,
+this change continues to mandate that the local CMC uses LRU eviction.
+
+This change specifies that during an eviction opportunity,
+the local CMC first checks a fixed number of entries from the tail of the LRU cache
+and evicts any TTL-expired entries,
+and then evicts as many entries from the tail as is necessary to not exceed the entry capacity.
+Limiting the number of entries checked helps to avoid latency spikes
+in the case that many entries expire around the same time.
+
+## Drawbacks
+
+Mandating that the local CMC support entry capacity values of 0 and 1,
+may result in edge cases in LRU removal logic
+(as hinted in the [C implementation](https://github.com/aws/aws-encryption-sdk-c/blob/aa85ca224d550cfe110e2112821a84506b9aca3e/source/local_cache.c#L924)).
+However, we believe that the functional benefits
+outweigh the complexity of the edge cases.
+
+By allowing the local CMC not to periodically evict TTL-expired entries,
+a scenario can arise in which
+an expired cache entry remains in the cache indefinitely
+because the cache is untouched after the entry enters.
+The specification represents a best-effort attempt to avoid retaining cache entries for long periods of time,
+and since this case is a relatively uncommon usage pattern,
+we are willing to accept this weakness.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any customer
+who relies on the local CMC behavior in the AWS Encryption SDK for C that,
+if the customer attempt to set an entry capacity of less than 2,
+instead sets the entry capacity to 2.
```

Comment:
```suggestion
who relies on the local CMC behavior in the AWS Encryption SDK for C,
if the customer attempt to set an entry capacity not equal to 2.
```
?

Reply:
What about just...

"who relies on the local CMC behavior in the AWS Encryption SDK for C
if the customer attempt to set an entry capacity less than 2"

?